include/marl/conditionvariable.h

```diff
@@ -114,15 +112,15 @@ void ConditionVariable::wait(std::unique_lock<std::mutex>& lock,
   if (auto fiber = Scheduler::Fiber::current()) {
     // Currently executing on a scheduler fiber.
     // Yield to let other tasks run that can unblock this fiber.
-    while (!pred()) {
-      mutex.lock();
-      waiting.emplace(fiber);
-      mutex.unlock();
-
-      lock.unlock();
-      fiber->yield();
-      lock.lock();
-    }
+    mutex.lock();
```

Comment:
For consistency?
```
{
     unique_lock() 
...
}
```

Reply:
I typically keep it as explicit lock / unlock if it's a one-liner.